File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/MenuManager.java
package com.supercilex.robotscouter.ui.teamlist;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

public interface MenuManager extends OnBackPressedListener {
    void onCreateOptionsMenu(Menu menu, MenuInflater inflater);

    /**
     * @see android.app.Activity#onOptionsItemSelected(MenuItem)
     */
    boolean onOptionsItemSelected(MenuItem item);

    void resetMenu();

    void saveState(Bundle outState);

    void restoreState(Bundle savedInstanceState);
}

File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/MenuManager.kt
package com.supercilex.robotscouter.ui.teamlist

import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem

interface MenuManager : OnBackPressedListener {
    fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater)

    /** @see android.app.Activity.onOptionsItemSelected */
    fun onOptionsItemSelected(item: MenuItem): Boolean

    fun resetMenu()

    fun saveState(outState: Bundle)

    fun restoreState(savedInstanceState: Bundle)
}
